### PR TITLE
Simplify `Arguments` in schema parser

### PR DESCRIPTION
**Pull request by tomhoule**
This is an opportunistic refactoring I made as I was trying to
understand the attribution validation code.

- Less code
- Fewer methods to call in consuming code
- Fewer allocations

### CODE REVIEW

File: libs/datamodel/core/src/transform/ast_to_dml/generator_loader.rs [hunk 1]
```diff
@@ -65,18 +65,25 @@ impl GeneratorLoader {
...
+            .as_str_from_env()?;
 
         let provider = StringFromEnvVar {
             name: PROVIDER_KEY,
             from_env_var,
             value,
         };
 
-        let output = if let Ok(arg) = args.arg(OUTPUT_KEY) {
+        let output = if let Some(arg) = args.get(OUTPUT_KEY) {
```
Comment (pimeys) [inline libs/datamodel/core/src/transform/ast_to_dml/generator_loader.rs:86]:
```rust
let output = if let Some(arg) = ...
```
always kind of looks weird, better with match...

File: libs/datamodel/core/src/transform/attributes/attribute_list_validator.rs [hunk 2]
```diff
@@ -61,14 +59,6 @@ impl<T: 'static> AttributeListValidator<T> {
                         ));
                     }
 
-                    if let Err(mut errs) = arguments.check_for_duplicate_named_arguments() {
```
Comment (pimeys) [inline libs/datamodel/core/src/transform/attributes/attribute_list_validator.rs:64]: where did we move these?

Comment (pimeys) [reply libs/datamodel/core/src/transform/attributes/attribute_list_validator.rs:64]: or is it the hashmap?

**Review by pimeys — APPROVED** [review]

File: libs/datamodel/core/src/transform/attributes/attribute_list_validator.rs (hunk 2 above)
Comment (tomhoule) [reply libs/datamodel/core/src/transform/attributes/attribute_list_validator.rs:64]: They're in the constructor now.

**Pull request merged**